File: Maternal_Mortality_Project/scripts/data_cleaning.py
import pandas as pd, numpy as np

df = pd.read_csv('../data/Pregnancy-Associated_Mortality.csv')
df['Live_Births'] = np.random.randint(20000, 40000, size=len(df))
df['MMR'] = (df['Deaths'] / df['Live_Births']) * 100000
df.to_csv('../data/maternal_mortality_clean.csv', index=False)

File: Maternal_Mortality_Project/scripts/data_cleaning.R


# Load required libraries
library(tidyverse)

# Load dataset
df <- read_csv("Pregnancy-Associated_Mortality.csv")

# Handle missing values
df <- df %>%
  drop_na(Year, `Race/ethnicity`) %>%
  mutate(Deaths = ifelse(is.na(Deaths), 0, Deaths))

# Standardize race categories
df <- df %>%
  mutate(`Race/ethnicity` = recode(`Race/ethnicity`,
    "Black, non-Hispanic" = "Black",
    "White, non-Hispanic" = "White",
    "Hispanic" = "Hispanic",
    "Asian/Pacific Islander" = "Asian/PI",
    .default = "Other/Unknown"
  ))

# Save cleaned dataset
write_csv(df, "maternal_mortality_clean.csv")
